### ISSUE TITLE
Set / Update Effort Estimation of Work Items

### ISSUE THREAD

**Issue by michaelkleinhenz** [fabric8io/fabric8-planner#662]
_From @michaelkleinhenz on September 29, 2016 11:19_

**Reference:** 1603E004
# Description

As a user, I want to see and update the effort estimation in the work item detail view.
# Functional Acceptance Criteria
1. The work item details dialog displays the effort estimate value.
2. The effort estimate field is a numeric field and defaults to 0.
3. The effort estimate field may not be empty.
4. Changes to the data are saved by clicking on the save button.
5. The effort estimate field behaviour is equal to the behaviour of the other fields in the view (see userstory #298, save behaviour, reset behaviour).

Note: the estimation is considered numeric, because in following stories, the estimation has to be numerically calculated with. Examples: calculating the total effort of an interation or measuring the team velocity. 
# Non-functional Acceptance Criteria

None.

_Copied from original issue: almighty/almighty-core#317_

**Comment by michaelkleinhenz** [comment]
Note: the nature of the field (numeric or other) is still under discussion. Points for numeric: we may have to sum up that value somewhere or find another way of calculating a "total effort value".

**Comment by michaelkleinhenz** [comment]
_From @Mgranfie on October 10, 2016 17:53_

@michaelkleinhenz E004 for the PDD reference does seem to fully align with this story? Can you let me know if this is the correct Experience or not?. Thanks...

**Comment by michaelkleinhenz** [comment]
_From @maxandersen on October 13, 2016 11:22_

@michaelkleinhenz I don't follow what a non-numeric field would mean for the estimation field here. Can you give an example ?

**Comment by michaelkleinhenz** [comment]
@maxandersen on the last bjn meeting on this(~3 weeks ago), there was discussion if the estimation is a numeric or a symbolic (s-m-l-xl-...) value. I think that does not make sense because we have to do calculations and diagramming with the estimation. I added the comment just to keep record that there was an unfinished discussions/concerns about it.

**Comment by michaelkleinhenz** [comment]
@Mgranfie you're right, E004 does not mention effort explicitly. Effort is only defined implicitly throughout the PDD and is only mentioned in the glossary. E004 is the one that comes at least near the area were the estimation should be :-)

**Comment by michaelkleinhenz** [comment]
_From @maxandersen on October 18, 2016 12:16_

> on the last bjn meeting on this(~3 weeks ago), there was discussion if the estimation is a numeric or a symbolic (s-m-l-xl-...) value

okey - but I just always understood that as behind the scenes the symbolic ones got mapped to a numerical value to actually give such reports ;)

**Comment by michaelkleinhenz** [comment]
@maxandersen yes, maybe. :-)

**Comment by Mgranfie** [comment]
@michaelkleinhenz as pat of the convo here am wondering if we need to display the unit of measure ie: hours, days, points? Has there been any discussions around this?

**Comment by Mgranfie** [comment]
@michaelkleinhenz should we also mention that any editing this value should be captured in the History?

**Comment by maxandersen** [comment]
> @maxandersen on the last bjn meeting on this(~3 weeks ago), there was discussion if the estimation is a numeric or a symbolic (s-m-l-xl-...) value. I think that does not make sense because we have to do calculations and diagramming with the estimation.

Just because the field is symbolic does not mean it can't be expressed as a numeric value nor that it can't be visualized. 

You would have things like high, medium, low that could map to 100, 50, 1. The absolute number might be useless though and not something you would want to add together and it is only useful for ordering.
